clouddriver-ecs/src/integration/java/com/netflix/spinnaker/clouddriver/ecs/test/CreateServerGroupWithArtifactsSpec.java

```diff
@@ -602,4 +601,121 @@ public void createServerGroup_ArtifactsEC2WithMultipleLBsTest()
         "integArtifactEC2TgMappingskWithMultipleLBsAndContainers-cluster",
         seenCreateServRequest.getCluster());
   }
+
+  @Test
+  public void createServerGroup_ProcessedArtifactsEC2TgMappingsTest()
+      throws IOException, InterruptedException {
+
+    // given
+    String url = getTestUrl(CREATE_SG_TEST_PATH);
+    String requestBody =
+        generateStringFromTestFile(
+            "/createServerGroup-spelProcessedArtifact-EC2-targetGroupMappings.json");
+    String expectedServerGroupName =
+        "ecs-integSpELProcessedArtifactsEC2TgMappingsStack-detailTest-v000";
+
+    String taskId =
+        given()
+            .contentType(ContentType.JSON)
+            .body(requestBody)
+            .when()
+            .post(url)
+            .then()
+            .statusCode(200)
+            .contentType(ContentType.JSON)
+            .body("id", notNullValue())
+            .body("resourceUri", containsString("/task/"))
+            .extract()
+            .path("id");
+
+    retryUntilTrue(
+        () -> {
+          List<Object> taskHistory =
+              get(getTestUrl("/task/" + taskId))
+                  .then()
+                  .contentType(ContentType.JSON)
+                  .extract()
+                  .path("history");
+          if (taskHistory
+              .toString()
+              .contains(String.format("Done creating 1 of %s", expectedServerGroupName))) {
+            return true;
+          }
+          return false;
+        },
+        String.format("Failed to detect service creation in %s seconds", TASK_RETRY_SECONDS),
+        TASK_RETRY_SECONDS);
+
+    ArgumentCaptor<RegisterTaskDefinitionRequest> registerTaskDefArgs =
+        ArgumentCaptor.forClass(RegisterTaskDefinitionRequest.class);
+    verify(mockECS).registerTaskDefinition(registerTaskDefArgs.capture());
+    RegisterTaskDefinitionRequest seenTaskDefRequest = registerTaskDefArgs.getValue();
+    assertEquals(expectedServerGroupName, seenTaskDefRequest.getFamily() + "-v000");
+    assertEquals(1, seenTaskDefRequest.getContainerDefinitions().size());
+    assertEquals(
+        "arn:aws:iam:::executionRole/testExecutionRole:1",
+        seenTaskDefRequest.getExecutionRoleArn());
+    assertEquals("application", seenTaskDefRequest.getContainerDefinitions().get(0).getName());
+    assertEquals(
+        "awslogs",
+        seenTaskDefRequest.getContainerDefinitions().get(0).getLogConfiguration().getLogDriver());
+    assertEquals(
+        "spinnaker-ecs-demo",
+        seenTaskDefRequest
+            .getContainerDefinitions()
+            .get(0)
+            .getLogConfiguration()
+            .getOptions()
+            .get("awslogs-group"));
+
+    assertEquals(
+        80,
+        seenTaskDefRequest.getContainerDefinitions().stream()
+            .filter(container -> container.getName().equals("application"))
```

Comment:
since there's only one container definition in this test, instead of filtering it out of the request each time (and incurring the cost of iterating over the list) we should retrieve it once and then do the assertions on that variable.

Reply:
make sense, changed.